test_pretty_next_run_with_todays_day_should_return_due_today breaks after midnight

### What happened?

The test `test_pretty_next_run_with_todays_day_should_return_due_today breaks` breaks when run after midnight. It does not seem increment the day.

### Module Version

3.2.2

### Device Type

Switcher V2

### Firmware Version

n/a

### Relevant log output

```shell
――――――――― test_pretty_next_run_with_todays_day_should_return_due_today ―――――――――

todays_day = <Days.WEDNESDAY: 'Wednesday'>, one_hour_from_now = '00:04'

    def test_pretty_next_run_with_todays_day_should_return_due_today(todays_day, one_hour_from_now):
        expected_return = f"Due today at {one_hour_from_now}"
>       assert_that(tools.pretty_next_run(one_hour_from_now, {todays_day})).is_equal_to(expected_return)
E       AssertionError: Expected <Due next Wednesday at 00:04> to be equal to <Due today at 00:04>, but was not.

tests/test_schedule_tools.py:52: AssertionError
```
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,6 +42,7 @@ pytest-mockservers = "^0.6.0"
 pytest-sugar = "^0.9.4"
 time-machine = "^2.7.0"
 yamllint = "^1.26.3"
+freezegun = ">=1.5.1"
 
 [tool.poetry.group.docs.dependencies]
 mkdocs = "^1.3.0"

--- a/src/aioswitcher/schedule/tools.py
+++ b/src/aioswitcher/schedule/tools.py
@@ -45,11 +45,15 @@ def pretty_next_run(start_time: str, days: Set[Days] = set()) -> str:
         current_datetime.time().strftime("%H:%M"), "%H:%M"
     ).time()
     schedule_time = datetime.strptime(start_time, "%H:%M").time()
-    current_time_plus_one_hour = (datetime.combine(datetime.today(), current_time) + timedelta(hours=1)).time()
+    current_time_plus_one_hour = (
+        datetime.combine(datetime.today(), current_time) + timedelta(hours=1)
+    ).time()
 
     execution_days = [d.weekday for d in days]
     # if scheduled for later on today, return "due today"
-    if current_weekday in execution_days and (current_time < schedule_time or current_time_plus_one_hour >= schedule_time):
+    if current_weekday in execution_days and (
+        current_time < schedule_time or current_time_plus_one_hour >= schedule_time
+    ):
         return f"Due today at {start_time}"
 
     execution_days.sort()

--- a/src/aioswitcher/schedule/tools.py
+++ b/src/aioswitcher/schedule/tools.py
@@ -45,10 +45,11 @@ def pretty_next_run(start_time: str, days: Set[Days] = set()) -> str:
         current_datetime.time().strftime("%H:%M"), "%H:%M"
     ).time()
     schedule_time = datetime.strptime(start_time, "%H:%M").time()
+    current_time_plus_one_hour = (datetime.combine(datetime.today(), current_time) + timedelta(hours=1)).time()
 
     execution_days = [d.weekday for d in days]
     # if scheduled for later on today, return "due today"
-    if current_weekday in execution_days and current_time < schedule_time:
+    if current_weekday in execution_days and (current_time < schedule_time or current_time_plus_one_hour >= schedule_time):
         return f"Due today at {start_time}"
 
     execution_days.sort()

--- a/tests/test_schedule_tools.py
+++ b/tests/test_schedule_tools.py
@@ -23,6 +23,7 @@ from assertpy import assert_that
 from pytest import fixture, mark
 
 from aioswitcher.schedule import Days, tools
+from freezegun import freeze_time
 
 days_by_weekdays = dict(map(lambda d: (d.weekday, d), Days))
 
@@ -48,6 +49,13 @@ def test_pretty_next_run_with_no_selected_days_should_return_due_today(one_hour_
 
 
 def test_pretty_next_run_with_todays_day_should_return_due_today(todays_day, one_hour_from_now):
+    expected_return = f"Due today at {one_hour_from_now}"
+    assert_that(tools.pretty_next_run(one_hour_from_now, {todays_day})).is_equal_to(expected_return)
+
+
+@freeze_time("23:00:00")
+def test_pretty_next_run_with_todays_day_end_of_day_should_return_due_today(todays_day):
+    one_hour_from_now = "00:00"
     expected_return = f"Due today at {one_hour_from_now}"
     assert_that(tools.pretty_next_run(one_hour_from_now, {todays_day})).is_equal_to(expected_return)
 

--- a/tests/test_schedule_tools.py
+++ b/tests/test_schedule_tools.py
@@ -20,10 +20,10 @@ from struct import pack, unpack
 
 import time_machine
 from assertpy import assert_that
+from freezegun import freeze_time
 from pytest import fixture, mark
 
 from aioswitcher.schedule import Days, tools
-from freezegun import freeze_time
 
 days_by_weekdays = dict(map(lambda d: (d.weekday, d), Days))
 

